agent/agent_test.go

```diff
@@ -1091,40 +1052,88 @@ func (s *AgentSuite) TestFetchProjectConfig() {
 	s.True(s.tc.privateVars["some_private_var"], "should include mock communicator private variables")
 }
 
-// getPanicLogs returns any panic-related mock logging.
-func (s *AgentSuite) getPanicLogs() []string {
-	var panicLogs []string
-	msgsByTask := s.mockCommunicator.GetMockMessages()
-	for _, msgs := range msgsByTask {
-		for _, msg := range msgs {
-			if strings.Contains(msg.Message, "panic") {
-				panicLogs = append(panicLogs, msg.Message)
-			}
-		}
-	}
-	return panicLogs
+func (s *AgentSuite) TestAbort() {
+	s.mockCommunicator.HeartbeatShouldAbort = true
+	s.a.opts.HeartbeatInterval = time.Millisecond
+
+	projYml := `
+buildvariants:
+- name: some_build_variant
+
+tasks:
+- name: this_is_a_task_name
+  commands:
+   - command: shell.exec
+     params:
+       script: sleep 5
+
+post:
+- command: shell.exec
```

Comment:
For the sake of completeness, I think it would be a good idea to add a test that makes sure that this also works for a teardown task without a post block.